src/site/resources/js/redirect.js

```diff
@@ -0,0 +1,39 @@
+window.onload = () => {
+    const currentUrl = window.location.href;
+    const urlObj = new URL(currentUrl);
+    const pathSegments = urlObj.pathname.split("/");
+    const lastSegment = pathSegments[pathSegments.length - 1];
+    const extractedUrl =  `/${lastSegment}${urlObj.hash}`;
+
+    if (currentUrl === "https://checkstyle.org/checks/") {
+        window.location.replace("https://checkstyle.org/checks");
+    }
+    else if (extractedUrl.match(/config_coding.html*/)) {
```

Comment:
Can we have map  with routing content "from-to"

"config_coding.html#VariableDeclarationUsageDistance" to "coding/variabledeclarationusagedistance.html#VariableDeclarationUsageDistance"

And we simply iterate over map entries and in case of match by endsWith choose destination.

so

Reply:
I have created a map which contains page title as key and its corresponding redirecting function as value, then used page title to get the redirect function from map and called it:

https://github.com/checkstyle/checkstyle/blob/c9a74285db34d2963947abf6effa07f610334335/src/site/resources/js/redirect.js#L12-L18


> And we simply iterate over map entries and in case of match by endsWith choose destination.

To use `endsWith` we will require to use if/else statement but I have used switch statements, using if/else has 2 downsides:

1. The code will become uglier, hard to read considering there will be a long list of if/else-if ladder
2. Switch statements are more efficient than if/else statements plus they make code look more clean.

So I think using switch statements is a good option for our usecase.

Reply:
The code is looking more cleaner and maintainable with current implementation.